lib/ruby_lsp/listeners/completion.rb

```diff
@@ -525,10 +532,139 @@ def complete_methods(node, name)
+                ),
+              )
+            end
+          end
+        return unless range
+
+        candidates.each do |param|
+          build_keyword_argument_completion_item(param, range)
+        end
+      end
+
+      #: (RubyIndexer::Entry::KeywordParameter | RubyIndexer::Entry::OptionalKeywordParameter param, Interface::Range range) -> void
+      def build_keyword_argument_completion_item(param, range)
+        new_text =
+          param.is_a?(RubyIndexer::Entry::KeywordParameter) ? param.decorated_name : "#{param.name}:".to_sym
+        new_text = @trigger_character == "," ? " #{new_text} " : "#{new_text} "
+        @response_builder << Interface::CompletionItem.new(
+          label: param.decorated_name,
+          text_edit: Interface::TextEdit.new(range: range, new_text: new_text),
+          kind: Constant::CompletionItemKind::PROPERTY,
+        )
+      end
+
+      #: (Prism::CallNode call_node, String? filter) -> Array[RubyIndexer::Entry::KeywordParameter | RubyIndexer::Entry::OptionalKeywordParameter]
+      def keyword_argument_completion_candidates(call_node, filter = nil)
+        method = resolve_method(call_node, call_node.message) if call_node
+        return [] unless method
+
+        signature = method.signatures.first
+        return [] unless signature
+
+        arguments = call_node.arguments&.arguments || []
+        keyword_arguments = arguments.find { _1.is_a?(Prism::KeywordHashNode) }
+
+        arg_names =
+          if keyword_arguments
+            keyword_arguments = T.cast(keyword_arguments, Prism::KeywordHashNode)
+            T.cast(
+              keyword_arguments.elements.select { _1.is_a?(Prism::AssocNode) },
+              T::Array[Prism::AssocNode],
+            ).map do |arg|
+              key = arg.key
+              arg_name =
+                case key
+                when Prism::StringNode then key.content
+                when Prism::SymbolNode then key.value
+                when Prism::CallNode then key.name
+                end
+
+              arg_name&.to_sym
+            end.compact
+          else
+            []
+          end
+
+        candidates = []
+        signature.parameters.each do |param|
+          unless param.is_a?(RubyIndexer::Entry::KeywordParameter) ||
+              param.is_a?(RubyIndexer::Entry::OptionalKeywordParameter)
+            next
+          end
+          # the argument is already provided
+          next if arg_names.include?(param.name)
+          # the argument is being typed halfway
+          next if filter && !param.name.start_with?(filter)
+
+          candidates << param
+        end
+
+        candidates
+      end
+
+      #: (Prism::CallNode node, String? name) -> (RubyIndexer::Entry::Member | RubyIndexer::Entry::MethodAlias)?
+      def resolve_method(node, name)
+        return unless name
+
+        type = @type_inferrer.send(:infer_receiver_for_call_node, node, @node_context)
+        return unless type
```

Comment:
Let's not rely on a private API. We can just call `@type_inferrer.infer_receiver_type(@node_context)` and make any necessary modifications to handle the new scenario in the type inferrer.